src/libraries/System.Private.CoreLib/src/System/Resources/ResourceReader.Core.cs

```diff
@@ -42,13 +45,22 @@ internal ResourceReader(Stream stream, Dictionary<string, ResourceLocator> resCa
             ReadResources();
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+            Justification = "InitializeBinaryFormatter will get trimmed out when AllowCustomResourceTypes is set to false. " +
+            "When set to true, we will already throw a warning for this feature switch, so we suppress this one in order for" +
+            "the user to only get one error.")]
         private object DeserializeObject(int typeIndex)
         {
             if (!_permitDeserialization)
             {
                 throw new NotSupportedException(SR.NotSupported_ResourceObjectSerialization);
             }
 
+            if (!AllowCustomResourceTypes)
```

Comment:
(nit) Might as well put this check first, since everything after it can be trimmed.